src/transformers_ocr.py

```diff
@@ -0,0 +1,382 @@
+#!/usr/bin/env python3
+# Copyright: Ren Tatsumoto <tatsu at autistici.org>
+# License: GNU GPL, version 3 or later; http://www.gnu.org/licenses/gpl.html
+
+import os
+import argparse
+from argparse import RawTextHelpFormatter
+import signal
+import shutil
+import sys
+import stat
+import subprocess
+import tempfile
+import dataclasses
+from typing import AnyStr, Collection, IO
+
+
+PROGRAM = "transformers_ocr"
+MANGA_OCR_PREFIX = os.path.join(os.environ["HOME"], ".local", "share", "manga_ocr")
+MANGA_OCR_PYENV_PATH = os.path.join(MANGA_OCR_PREFIX, "pyenv")
+MANGA_OCR_PYENV_PIP_PATH = os.path.join(MANGA_OCR_PYENV_PATH, "bin", "pip")
+
+PIPE_PATH = "/tmp/manga_ocr.fifo"
+PID_FILE = "/tmp/manga_ocr.pid"
+
+SPACE = "、"
+CONFIG_PATH = os.path.join(
+    os.environ.get("XDG_CONFIG_HOME", os.path.join(os.environ["HOME"], ".config")),
+    "transformers_ocr",
+    "config",
+)
+
+
+def get_clip_copy_args():
+    if is_Xorg():
+        return (
+            "xclip",
+            "-selection",
+            "clipboard",
+        )
+    return ("wl-copy",)
+
+
+def get_platform():
+    if is_Xorg():
+        return "Xorg"
+    elif is_GNOME():
+        return "GNOME"
+    else:
+        return "Wayland"
+
+
+def is_Xorg():
+    return "WAYLAND_DISPLAY" not in os.environ
+
+
+def is_GNOME():
+    return os.environ.get("XDG_CURRENT_DESKTOP") == "GNOME"
+
+
+def if_installed(*programs):
+    for prog in programs:
+        if (
+            not shutil.which(prog)
+            and subprocess.call(
+                ["pacman", "-Qq", prog],
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL,
+            )
+            != 0
+        ):
+            notify_send(f"{prog} must be installed for {PROGRAM} to work.")
+            sys.exit(1)
+
+
+def take_screenshot(screenshot_path):
+    if is_Xorg():
+        if_installed("maim", "xclip")
+        subprocess.run(
+            [
+                "maim",
+                "--select",
+                "--hidecursor",
+                "--format=png",
+                "--quality",
+                "1",
+                screenshot_path,
+            ],
+            check=True,
+        )
+    elif is_GNOME():
+        if_installed("gnome-screenshot", "wl-copy")
+        subprocess.run(["gnome-screenshot", "-a", "-f", screenshot_path], check=True)
+    else:
+        if_installed("grim", "slurp", "wl-copy")
+        subprocess.run(
+            [
+                "grim",
+                "-g",
+                subprocess.check_output(["slurp"]).decode().strip(),
+                screenshot_path,
+            ],
+            check=True,
+        )
+
+
+def prepare_pipe():
+    if os.path.exists(PIPE_PATH):
+        os.unlink(PIPE_PATH)
+    if not os.path.exists(PIPE_PATH):
+        os.mkfifo(PIPE_PATH)
```

Comment:
The original function looked like this:
```
def prepare_pipe():
    if os.path.isfile(PIPE_PATH):
        os.remove(PIPE_PATH)
    if not is_fifo(PIPE_PATH):
        os.mkfifo(PIPE_PATH)
```
This function will delete the pipe every time it is run.

Reply:
I have replaced the function with the original version because otherwise it would cause freezes.